### PR TITLE
pinephone-uboot: update to 2024.07

### PR DESCRIPTION
**Pull request by TheStilbruch**
#### Testing the changes
- I tested the changes in this PR: **YES**

#### Local build testing
- I built this PR locally for these architectures (if supported. mark crossbuilds):
  - aarch64-musl

Tested this on my pinephone and everything booted up properly. This is my first contribution to the project, let me know if I need to fix anything.

### CODE REVIEW

File: srcpkgs/pinephone-uboot/template [hunk 1]
```diff
@@ -1,32 +1,32 @@
 # Template file for 'pinephone-uboot'
 pkgname=pinephone-uboot
-version=0.0.20200917
+version=0.0.202404
 revision=3
```
Comment (Eloitor) [inline srcpkgs/pinephone-uboot/template:4]: revision=1

File: srcpkgs/pinephone-uboot/template [hunk 2]
```diff
@@ -1,32 +1,32 @@
...
 conf_files="/etc/default/${pkgname}-config"
 short_desc="U-Boot for Pinephone with Crust support"
 maintainer="John Sullivan <jsullivan@csumb.edu>"
 license="GPL-2.0-only, BSD-3-Clause"
 homepage="http://www.denx.de/wiki/U-Boot/WebHome"
 
-_tag_uboot=crust-2020-09-17
-_ver_armtf=2.4
-_ver_crust=0.3
+_ver_uboot=2024.04
```
Comment (Eloitor) [inline srcpkgs/pinephone-uboot/template:15]: Isn't the latest version 2024.07?

**Comment by TheStilbruch** [comment]
Comments have been addressed in this latest commit

**Comment by Eloitor** [comment]
Hi, I tried to build an image using the pinephone-uboot on void's repositories using

```
cd void-mklive
sudo ./mkrootfs.sh aarch64-musl
sudo ./mkplatformfs.sh -p "mesa xfce4" pinephone void-aarch64-musl-ROOTFS-$(date +%Y%m%d).tar.xz
sudo ./mkimage.sh void-pinephone-PLATFORMFS-$(date +%Y%m%d).tar.xz
```

I also tried without `-musl`. I was not able to boot with neither of the two images on eMMC nor on SD card. What I'm doing wrong? Once I can boot one of these images I'll test this PR in my pinephone.

This is what I get when building the musl image:
```
Configuring image for platform pinephone
qemu-aarch64-static: /lib/ld-musl-aarch64.so.1: Invalid ELF image for this architecture
```
The glib image does not give any error, but it does not boot.

**Comment by Eloitor** [comment]
maybe a `srcpkgs/pinephone-uboot/update` file can be made, fetching the data from https://docs.u-boot.org/en/latest/develop/release_cycle.html#previous-releases

**Comment by TheStilbruch** [comment]
> maybe a `srcpkgs/pinephone-uboot/update` file can be made, fetching the data from https://docs.u-boot.org/en/latest/develop/release_cycle.html#previous-releases

Good idea, I'll look into that now.

**Comment by Eloitor** [comment]
Once you install the image to the pinephone can you resize it? I installed it to eMMC but if I resize it to use the full space it no longer boots...

**Comment by TheStilbruch** [comment]
> Once you install the image to the pinephone can you resize it? I installed it to eMMC but if I resize it to use the full space it no longer boots...

I'm not sure, I've been installing the image to the SD card and booting from that.  Found  [this link](https://git.toradex.com/cgit/meta-toradex-bsp-common.git/tree/recipes-core/fs-init/files/resizefs.sh?h=morty) which looks like it's trying to do the same thing?

**Comment by TheStilbruch** [comment]
> maybe a `srcpkgs/pinephone-uboot/update` file can be made, fetching the data from https://docs.u-boot.org/en/latest/develop/release_cycle.html#previous-releases

I am not sure I understand the syntax of these update files too well. The manual page explains it some but not enough to do what I want. 

Here's what I have

```conf
site="https://docs.u-boot.org/en/latest/develop/release_cycle.html"
pkgname="u-boot"
pattern="v\K(\d+)\.(\d+)(?=</a>)"
```

**Comment by Eloitor** [comment]
I tested it on the pinephone and it worked :)

**Comment by TheStilbruch** [comment]
I've modified the version number to match upstream uboot, and the `u-boot-tools` package. The update script is working as well. Should be all ready to merge now.

**Comment by Eloitor** [comment]
Are you interested in updating this PR for 2025.01 ?

**Pull request closed without merging**